src/bindings/python/fluxacct/accounting/visuals.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+###############################################################
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+import shutil
+import sys
+
+
+class RenderBarGraph:
+    """A class for displaying a horizontal bar graph."""
+
+    def __init__(self, conn, query, query_params=(), bar_char="█", fallback_char="#"):
+        """
+        Initialize the graph renderer.
+
+        Parameters:
+            conn: A SQLite Connection object.
+            query: The query to the flux-accounting DB to actually fetch the data used
+                in the graph.
+            query_params: Any parameters specified for the SQLite query.
+            bar_char: The primary character to draw the bars (default: block char).
+            fallback_char: A fallback ASCII character if primary char can't be encoded in
+                the terminal.
+        """
+        self.conn = conn
+        self.bar_char = bar_char
+        self.fallback_char = fallback_char
+        self.query = query
+        self.query_params = query_params
+
+    def fetch_rows(self):
+        cur = self.conn.cursor()
+        if self.query_params != ():
+            cur.execute(self.query, self.query_params)
+        else:
+            cur.execute(self.query)
+        return cur.fetchall()
+
+    def make_label(self, row):
+        """Return string label for a row; override in subclasses."""
+        raise NotImplementedError
+
+    def _bar_char(self):
+        """Return the appropriate bar character based on terminal encoding support."""
+        encoding = sys.stdout.encoding or "utf-8"
+        try:
+            self.bar_char.encode(encoding)
+            return self.bar_char
+        except (UnicodeEncodeError, TypeError):
+            return self.fallback_char
+
+    def draw(self):
+        rows = self.fetch_rows()
+        if not rows:
+            return "no data to display"
+
+        labels = [self.make_label(row) for row in rows]
+        # last column will be the numerical value to display
+        values = [row[-1] for row in rows]
+
+        term_width = shutil.get_terminal_size((80, 20)).columns
```

Comment:
It looks like the default for `get_terminal_size` is `(80,24)`, and you don't check the number of rows so I think you could remove the argument to this function?